runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -837,11 +837,15 @@ public void onProcessingTime(InternalTimer<ByteBuffer, TimerData> timer) {
 
   // allow overriding this in ExecutableStageDoFnOperator to set the key context
   protected void fireTimerInternal(Object key, TimerData timerData) {
```

Comment:
```suggestion
  protected void fireTimerInternal(ByteBuffer key, TimerData timerData) {
```

Reply:
Can we change this to `ByteBuffer` now? Same in `ExecutableStageDoFnOperator` where it is overridden.